packages-content-model/roosterjs-content-model-types/lib/format/formatParts/BorderFormat.ts

```diff
@@ -26,4 +26,16 @@ export type BorderFormat = {
      * Radius to be applied in all borders corners
      */
     borderRadius?: string;
+
+    /**
+     * Inline Start border in format 'width style color'
+     * Left border in Left to right languages, right border in right to left languages
+     */
+    borderInlineStart?: string;
```

Comment:
consider remove original properties `borderLeft` and `borderRight` to keep "single source of truth"

Reply:
We are using `borderLeft` and `borderRight`, if you remove this properties, please test the impact in the image border feature.

Reply:
I just did some test, and I found that borderLeft/Right and inlineBorderStart/End can have different values.

When they are conflict, inlineBorderStart/End will overwrite borderLeft/Right when render
When they are not conflict, both take effect.

This cause a problem that if we use inlineBorderStand/End, we also need to take care of borderLeft/Right, which means if there is already borderLeft/Right, when we set inlineBorderStart/End, it may not fully overwrite the original one. For example:

```html
<div style="border-left: solid 1px red; border-inline-start: solid 10px red; direction: rtl">test</div>
```

Because this DIV is in RTL, border-inline-start and border-left mean different thing, so we will see two borders on left and right. To solve this, we need to know if the DIV is in LTR or RTL when get/set border. That makes the benefit of using inline-border-start/end are gone.

In fact, in some format APIs, we are reading direction and do different handling for direction, for example: https://github.com/microsoft/roosterjs/blob/9aafcb65241bf936cc0cf12dbbb1f4ab5c827547/packages-content-model/roosterjs-content-model-api/lib/modelApi/block/setModelIndentation.ts#L53

And for some other cases, we are doing a translate of the property value in format handler to let the property not direction-sensitive, for example: https://github.com/microsoft/roosterjs/blob/9aafcb65241bf936cc0cf12dbbb1f4ab5c827547/packages-content-model/roosterjs-content-model-dom/lib/formatHandlers/utils/dir.ts#L30

The second one looks more elegant, the down side is it makes the related format need to be combined with DirectionFormat: https://github.com/microsoft/roosterjs/blob/9aafcb65241bf936cc0cf12dbbb1f4ab5c827547/packages-content-model/roosterjs-content-model-dom/lib/formatHandlers/block/textAlignFormatHandler.ts#L9

![image](https://github.com/microsoft/roosterjs/assets/23065085/8c84a8c0-5d64-49f0-a906-92f985c4d62b)

@Andres-CT98 I think either way above can work here. What do you think?

Reply:
How does the example div happen in the editor? Using our Insert table functions all tables should have the correct side border properties, and we could handle paste cases. If the example div is willingly inserted on the editor as HTML then the behaviour is as expected.

Reply:
Possibly load existing HTML from draft.

Reply:
Ok, I think I can make the change in borderFormatHandler, however I do not know how it works or when is it called. Ideally all instances of borderLeft/Right would be changed accordingly.

There is also a recent change with border widths that I have to change as well, but I could not find out why that change was made.

Reply:
`*FormatHandler.apply` is called when we write Content Model back into editor DOM.

For example, when there is a DIV with border-left to be red, and direction is ltr, we read them and store the value as
```ts
// borderFormatHandler.parse
format.borderStart = 'red'
```

Then in a format API we change its direction to rtl, and border-start to green:

```ts
format.direction = 'rtl';
format.borderStart = 'gren';
```

Then we write it back, the borderFormatHandler.apply will be called and we will write border-right instead of border-left since we know we are in rtl now:
```ts
// borderFormatHandler.apply
if (format.direction == 'rtl') {
   element.style.borderRight = format.borderStart;
   element.style.borderLeft = format.borderEnd;
}  else {
   element.style.borderLeft = format.borderStart;
   element.style.borderRight = format.borderEnd;
}
```

That makes the direction is totally separated from format API. In format API we only need to care borderStart/End, no need to know if it is ltr or rtl. They will always be handled by formatHandler.

Reply:
After trying the handler approach, I realised that the issue was both in the application of the borders, and with the RTL direction change itself.